kong/clustering/rpc/concentrator.lua

```diff
@@ -153,45 +226,44 @@ function _M:_event_loop(lconn)
           local reply_to = assert(call.reply_to,
                                   "unknown requester for RPC")
 
-          local res, err = self.manager:_local_call(target_id, payload.method,
-                                                    payload.params, not payload.id)
+          if not isarray(payload) then
+            -- one rpc call
+            self:process_one_request(target_id, reply_to, payload)
 
-          -- notification has no callback or id
-          if not payload.id then
-            ngx_log(ngx_DEBUG, "[rpc] notification has no response")
             goto continue
           end
 
-          if res then
-            -- call success
-            res, err = self:_enqueue_rpc_response(reply_to, {
-              jsonrpc = jsonrpc.VERSION,
-              id = payload.id,
-              result = res,
-            })
+          -- rpc call with an empty Array
+          if isempty(payload) then
+            local res, err = self:_enqueue_rpc_response(
+                              reply_to,
+                              new_error(nil, jsonrpc.INVALID_REQUEST, "Invalid Request"))
```

Comment:
more meaningful error message instead of "Invalid Request"?

Reply:
Currently we only have this, see https://www.jsonrpc.org/specification#response_object

Reply:
Can we replace the "Invalid Request" with something like `Batch call without any request object", I think the error.message can be customized according to the JSON RPC Spec

Reply:
Of cause, but it needs that koko must understand these new error code, perhaps it is out of this PR's scope.

Reply:
msg changed.